x-pack/plugins/security_solution/public/detection_engine/rule_management/components/rule_details/three_way_diff/final_edit/fields/threshold/threshold_adapter.tsx

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+import { EuiAutoSizer } from '@elastic/eui';
+import { useDiffableRuleDataView } from '../hooks/use_diffable_rule_data_view';
+import type { DiffableRule } from '../../../../../../../../../common/api/detection_engine';
+import { ThresholdEdit } from '../../../../../../../rule_creation/components/threshold_edit';
+
+const MIN_ROW_LAYOUT_WIDTH = 560;
+
+interface ThresholdAdapterProps {
+  finalDiffableRule: DiffableRule;
+}
+
+export function ThresholdAdapter({ finalDiffableRule }: ThresholdAdapterProps): JSX.Element {
+  const { dataView } = useDiffableRuleDataView(finalDiffableRule);
+
+  const esFields = dataView?.fields || [];
+
+  return (
+    <EuiAutoSizer disableHeight>
```

Comment:
Have you tried resolving layout issues via flex or grid? It looks less efficient using an auto sizer here.

Reply:
Updated this part. Used container queries instead.